javascript/ql/src/semmle/javascript/Closure.qll

```diff
@@ -267,6 +267,9 @@ module Closure {
       result = this
     }
 
-    override DataFlow::Node getBoundReceiver() { result = getArgument(1) }
+    override DataFlow::Node getBoundReceiver(DataFlow::Node callback) {
```

Comment:
Here, and for all of the other new `getBoundReceiver` overrides: we should keep the existing `getBoundReceiver`overrides to preserve backwards compatibility. As it is now, the meaning for external uses of `getBoundReceiver()` is `none()` rather than `result = getArgument(1)`.

Reply:
See the implementation of `PartialInvokeNode.getBoundReceiver`, it dispatches to the one-argument version:
```
class PartialInvokeNode {
    DataFlow::Node getBoundReceiver() { result = range.getBoundReceiver(_) }
}
```
So external calls to `getBoundReceiver()` will indeed get the same result.

However, existing overrides of `Range.getBoundReceiver()` won't be seen by calls to `PartialInvokeNode.getBoundReceiver()`. I couldn't find a way to maintain both as we can't call the deprecated method ourselves.